### PR TITLE
Add pipeline for `pydantic-next`

### PR DESCRIPTION
**Pull request by Kludex**
- Replaces https://github.com/pydantic/pydantic/pull/5592
- Closes https://github.com/pydantic/pydantic/issues/5465

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -289,6 +289,35 @@ jobs:
...
+
+    - name: install deps
+      run: pip install -r requirements/pyproject-all.txt -r requirements/testing-extra.txt
+
+    - name: install pydantic next
+      working-directory: pydantic-next
+      run: pip install .
+
+    - name: replace imports on test
+      run: find tests -type f -name '*.py' -exec sed -i '' -E 's/(import|from) pydantic([\. ])/\1 pydantic_next\2/g' {} +
```
Comment (Kludex) [inline .github/workflows/ci.yml:312]:
```suggestion
      run: find tests -type f -name '*.py' -exec sed -i '' -E 's/(import|from) pydantic([^_]|$)/\1 pydantic_next\2/g' {} +
```

File: tests/test_migration.py [hunk 2]
```diff
@@ -1,10 +1,15 @@
 import importlib
+import sys
 
 import pytest
 
 from pydantic._migration import DEPRECATED_MOVED_IN_V2, MOVED_IN_V2, REMOVED_IN_V2
 from pydantic.errors import PydanticImportError
 
+pytestmark = pytest.mark.skipif(
+    'pydantic_next' in sys.modules, reason="The migration logic doesn't make sense for pydantic_next"
```
Comment (Kludex) [inline tests/test_migration.py:10]:
```suggestion
    'pydantic_next' in sys.modules, reason='The migration logic doesn't make sense for pydantic_next'
```

File: tests/test_migration.py [hunk 3]
```diff
@@ -1,10 +1,15 @@
 import importlib
+import sys
 
 import pytest
 
 from pydantic._migration import DEPRECATED_MOVED_IN_V2, MOVED_IN_V2, REMOVED_IN_V2
 from pydantic.errors import PydanticImportError
 
+pytestmark = pytest.mark.skipif(
+    'pydantic_next' in sys.modules, reason='The migration logic doesn't make sense for pydantic_next'
```
Comment (Kludex) [inline tests/test_migration.py:10]:
```suggestion
    'pydantic_next' in sys.modules, reason="The migration logic doesn't make sense for pydantic_next"
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -289,6 +289,35 @@ jobs:
...
+
+    - name: install deps
+      run: pip install -r requirements/pyproject-all.txt -r requirements/testing-extra.txt
+
+    - name: install pydantic next
+      working-directory: pydantic_next
+      run: pip install .
+
+    - name: replace imports on test
+      run: find tests -type f -name '*.py' -exec sed -i '' -E 's/(import|from) pydantic([^_]|$)/\1 pydantic_next\2/g' {} +
```
Comment (Kludex) [inline .github/workflows/ci.yml:312]:
```suggestion
      run: find tests -type f -name '*.py' -exec sed -E 's/(import|from) pydantic([^_]|$)/\1 pydantic_next\2/g' {} > tmpfile && mv tmpfile {} \;
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -289,6 +289,35 @@ jobs:
...
+
+    - name: install deps
+      run: pip install -r requirements/pyproject-all.txt -r requirements/testing-extra.txt
+
+    - name: install pydantic next
+      working-directory: pydantic_next
+      run: pip install .
+
+    - name: replace imports on test
+      run: find tests -type f -name '*.py' -exec sed -E 's/(import|from) pydantic([^_]|$)/\1 pydantic_next\2/g' {} > tmpfile && mv tmpfile {} \;
```
Comment (Kludex) [inline .github/workflows/ci.yml:312]:
```suggestion
      run: |
          find tests -type f -name '*.py' -exec \
              sed -E 's/(import|from) pydantic([^_]|$)/\1 pydantic_next\2/g' {} > tmpfile && \
              mv tmpfile {} \;
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -289,6 +289,38 @@ jobs:
...
+
+    - name: install pydantic next
+      working-directory: pydantic_next
+      run: pip install .
+
+    - name: replace imports on test
+      run: |
+          find tests -type f -name '*.py' -exec \
+              sed -E 's/(import|from) pydantic([^_]|$)/\1 pydantic_next\2/g' {} > tmpfile && \
+              mv tmpfile {} \;
```
Comment (Kludex) [inline .github/workflows/ci.yml:315]:
```suggestion
           find tests -type f -name '*.py' -exec sed -i -E 's/(import|from) pydantic([^_]|$)/\1 pydantic_next\2/g' {} +
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -286,6 +286,48 @@ jobs:
           allowed-skips: test-fastapi  # TODO: drop once re-enabled
           jobs: ${{ toJSON(needs) }}
 
+  publish_next:
```
Comment (Kludex) [inline .github/workflows/ci.yml:289]: @samuelcolvin The idea is for this job to only update the branch, and then `release` job would release both?

**Comment by samuelcolvin** [comment]
after a long discussion, we decided to embed pydantic V1 inside V2, instead of doing this #5771.

Sorry for the change of plan.

**Pull request closed without merging**